common/utils.go

```diff
@@ -18,3 +23,83 @@ func ErrorMessages(errors []error) string {
 	}
 	return strings.Join(result, "\n")
 }
+
+func WithNetNS(ns netns.NsHandle, work func() error) error {
```

Comment:
To `vishvananda/netns`? I can do that async with this PR getting merged, and then can submit another PR here to update the vendored dependency when it makes it upstream.